src/linear_algebra/char_poly/coeff.lean

```diff
@@ -156,6 +156,27 @@ begin
 end
 
 variables {p : ℕ} [fact p.prime]
+#where
+-- the following is a nasty case bash that I (DT) abstracted from the lemma below
+-- a pre-existing comment asked "maybe it can be proven more... algebraically?"
+lemma mat_poly_equiv_eq_X_pow_sub_C {K : Type*} (p k : ℕ) [field K] (M : matrix n n K) :
+  mat_poly_equiv
+      (((expand K (p ^ k) : polynomial K →+* polynomial K).map_matrix) (char_matrix (M ^ p ^ k))) =
```

Comment:
can trim some parens:
```suggestion
      ((expand K (p ^ k) : polynomial K →+* polynomial K).map_matrix (char_matrix (M ^ p ^ k))) =
```

Reply:
I will wait to accept this, so that I see more quickly if something else does not build.

Reply:
This one is also in!